src/python/gudhi/datasets/remote.py

```diff
@@ -0,0 +1,199 @@
+    This folder is used to store remotely fetched datasets.
+    By default the datasets directory is set to a folder named 'remote_datasets' in the user home folder.
+    Alternatively, it can be set by giving an explicit folder path. The '~' symbol is expanded to the user home folder.
+    If the folder does not already exist, it is automatically created.
+
+    Parameters
+    ----------
+    data_home : string
+        The path to remote datasets directory. Default is `None`, meaning that the data home directory will be set to "~/remote_datasets".
+
+    Returns
+    -------
+    data_home: string
+        The path to remote datasets directory.
+    """
+    if data_home is None:
+        data_home = join("~", "remote_datasets")
+    data_home = expanduser(data_home)
+    makedirs(data_home, exist_ok=True)
+    return data_home
+
+
+def clear_data_home(data_home = None):
+    """
+    Delete all the content of the data home cache.
+
+    Parameters
+    ----------
+    data_home : string, default is None.
+        The path to remote datasets directory. If `None`, the default directory to be removed is set to "~/remote_datasets".
+    """
+    data_home = get_data_home(data_home)
+    shutil.rmtree(data_home)
+
+def _checksum_sha256(file_path):
+    """
+    Compute the file checksum using sha256.
+
+    Parameters
+    ----------
+    file_path: string
+        Full path of the created file.
+
+    Returns
+    -------
+        The hex digest of file_path.
+    """
+    sha256_hash = hashlib.sha256()
+    chunk_size = 4096
+    with open(file_path,"rb") as f:
+        # Read and update hash string value in blocks of 4K
+        while True:
+            buffer = f.read(chunk_size)
+            if not buffer:
+                break
+            sha256_hash.update(buffer)
+    return sha256_hash.hexdigest()
+
+def _fetch_remote(url, filename, dirname = "remote_datasets", file_checksum = None, accept_license = False):
+    """
+    Fetch the wanted dataset from the given url and save it in file_path.
+
+    Parameters
+    ----------
+    url : string
+        The url to fetch the dataset from.
+    filename : string
+        The name to give to downloaded file.
+    dirname : string
+        The directory to save the file to. Default is "remote_datasets".
+    file_checksum : string
+        The file checksum using sha256 to check against the one computed on the downloaded file.
+        Default is 'None'.
+    accept_license : boolean
+        Flag to specify if user accepts the file LICENSE and prevents from printing the corresponding license terms.
+        Default is False.
+
+    Returns
+    -------
+    file_path: string
+        Full path of the created file.
+    """
+
+    file_path = join(dirname, filename)
+
+    # Get the file
+    urlretrieve(url, file_path)
+
+    if file_checksum is not None:
+        checksum = _checksum_sha256(file_path)
+        if file_checksum != checksum:
```

Comment:
Hmm, we could delete it indeed since it is cleaner, but I am wondering if leaving it could help the user debugging...

Reply:
Part of the question is what happens when a download fails (bad internet). Is there a risk we may end up with an incomplete / broken file in the right position, which we don't remove, and the next time we call fetch we don't actually retry the download because there is already a file there (and we actually try to use that file without checking the checksum)? Then removing looks better.
Or does urlretrieve download with a different name and only atomically renames at the end in case of success? Then in case we have a bug where we provided a wrong checksum, not removing avoids redownloading the file many times for nothing when there was nothing wrong with the first download.

The main issue to me is if we can easily end up loading a file even though we noticed that the checksum didn't match. Removing wrong files looks like one easy possible way to reduce this probability, but others are possible if you prefer.

Reply:
Yeah, removing the file sounds safer.